src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1482,15 +1514,40 @@ private void ImportCasting(ILOpcode opcode, int token)
             else
                 function = throwing ? "CheckCastClass" : "IsInstanceOfClass";
 
-            var arguments = new StackEntry[]
+            StackEntry[] arguments;
+            if (type.IsRuntimeDeterminedSubtype)
             {
-                new LoadExpressionEntry(StackValueKind.ValueType, "eeType", GetEETypePointerForTypeDesc(type, true), _compilation.TypeSystemContext.SystemModule.GetKnownType("System", "EETypePtr")),
-                _stack.Pop()
-            };
+                //TODO refactor argument creation with else below
+                arguments = new StackEntry[]
+                            {
+                                new ExpressionEntry(StackValueKind.ValueType, "eeType", CallGenericHelper(ReadyToRunHelperId.TypeHandle, type),
```

Comment:
This would probably fix the `TestNullableCasting` test in the `Simple\Generics` test:

```suggestion
                                new ExpressionEntry(StackValueKind.ValueType, "eeType", CallGenericHelper(ReadyToRunHelperId.TypeHandleForCasting, type),
```

If it doesn't fix it, then leave this as-is for now. Let's not add more scope to this.